Version_Control_Exercise.md

```diff
@@ -0,0 +1,183 @@
+Now, **Person 2** should contribute their new file via the normal process (add, commit, push). For example, let's say you had created `person2.txt`. You would run:
+
+    git add person2.txt
+    git commit -m "create person 2 file"
+    git push
+
+Next, **Person 3** should attempt the same. Everything works up until the push, where you'll get an error message. (Go ahead and try, so you can see the message.) That's because git does not allow you to push if you're not up-to-date with the remote. We can fix this.
+
+## Reconciling Diverged Branches via Merging
+To handle diverging branches, we will perform a "merge commit." With Person 2 having successfully pushed, but Person 3 unable to do so, we will need to resolve this. We can do this with the following steps:
+
+ 1. **Person 3** should run `git fetch`. Afterwards, `git status` should show diverged branches.
+ 2. Now, **Person 3** should run `git merge`. This will create a new "merge commit" following from both diverged paths and bringing them back together. It will also pop open the terminal text editor (either `vim` or `nano`) to allow you to edit a message for this merge. 
+ 3. For now, just keep the default and close the editor. For `nano`, press ctrl+x to close out. For `vim`, hit the escape key, and then type `:q` and hit the enter key.
+ 4. Now, **Person 3** should make sure to push and ***everyone else*** should pull.
+
+The changes have been successfully reconciled across different files! However, we still have one scenario to explore: What happens when two people edit the same file?
+
+## Resolving Merge Conflicts
+There's one more issue to consider when reconciling divergent commits - what if the commits are made to the same file, including potential contradictory changes!?
+
+In this case, you've got a merge conflict. Regardless of whether you choose to rebase or merge, git is going to need some help figuring out what to do. For our purposes, we will not be covering rebasing (as rewriting commit history is bad) and only focus on merging. In fact, for the purpose of MRover, ***you should almost never need to rebase***.
+
+For this section, let **Person 1** and **Person 2** lead. Both should make sure you are up to date with the changes from the last section.
+
+First, have **Person 2** edit the `README.md` copy, as such:
+
+    # mrover-vc-exercise Person 2 was here.
+    Our members in our group are Zachary Goldston, Cameron Tressler, and Ashwin Gupta.
+    Also changing stuff here.
+
+Additionally, make a new files called `new.txt`. Put whatever text you want in it.
+
+Now, **Person 2** should contribute their changes. The commands are as follows:
+
+    git add -A
+    git commit -m "person 2 changes"
+    git push
+
+Note that the `-A` flag will stage all files that have been modified.
+
+Now, **Person 1** should make one change to their copy of `README.md`.
```

Comment:
Hopefully they get the point, but just in case, probably good to specify this should happen before pulling

Reply:
Good call